### PR TITLE
Add a getDeviceByLabel method

### PR DESCRIPTION
**Pull request by agrare**
Add a method to return a VirtualDevice by label and refactor the
existing getDeviceKeysByLabel to use it.

https://bugzilla.redhat.com/show_bug.cgi?id=1533728

### CODE REVIEW

File: lib/VMwareWebService/MiqVimVm.rb [hunk 1]
```diff
@@ -940,18 +940,14 @@ def getDeviceByBacking(backingFile, hardware = nil)
...
-    # controller_key, key, unit_number
-    [nil, nil, nil]
+  def getDeviceByLabel(device_label, hardware = nil)
+    hardware ||= getHardware
+    hardware["device"].to_a.detect { |dev| dev["deviceInfo"]["label"] == device_label }
+  end
+
+  def getDeviceKeysByLabel(device_label, hardware = nil)
+    dev = getDeviceByLabel(device_label, hardware)
+    dev.values_at("controllerKey", "key", "unitNumber") unless dev.nil?
```
Comment (roliveri) [inline lib/VMwareWebService/MiqVimVm.rb]: @agrare 
Is it ok that this no longer returns an array when no devices are found?

Comment (agrare) [reply lib/VMwareWebService/MiqVimVm.rb]: Yes, the only caller of this is https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb#L338 which with this changeset applied results in:
```
>> controller_key, key, unit_number = miq_vim_vm.send(:getDeviceKeysByLabel, "abcd")
=> nil
>> controller_key
=> nil
>> key
=> nil
>> unit_number
=> nil
```

**Comment by agrare** [comment]
@roliveri WDYT?

**Pull request merged**